books.js

```diff
@@ -1,18 +1,137 @@
+/* eslint-disable no-unused-vars */
+class Book2 {
+  constructor(title, author, next_node = null) {
     this.title = title;
     this.author = author;
   }
 }
 
+class BookList2 {
+  constructor() {
+    this.head = null;
+    this.tail = null;
+  }
+
+  add(title, author) {
+    const newnode = new Book2(title, author);
+    if (this.tail === null) {
+      this.head = newnode;
+      this.tail = newnode;
+    } else {
+      this.tail.next_node = newnode;
+      this.tail = newnode;
+    }
+  }
+
+  size() {
+    let count = 0;
+    let node = this.head;
+    while (node) {
+      count += 1;
+      node = node.next_node;
+    }
+    return count;
+  }
+
+  print() {
+    let nodes = this.head;
+    while (nodes) {
+      nodes = nodes.next_node;
+      return `${nodes.title} ${nodes.author}`;
+    }
+    return null;
+  }
+
+  printArray() {
+    const arr = [];
+    let node = this.head;
+    while (node) {
+      const obj = {
+        title: node.title,
+        author: node.author,
+      };
+      arr.push(obj);
+      node = node.next_node;
+    }
+    return arr;
+  }
+
+  get(index) {
+    let node = this.head;
+    let count = 0;
+    while (node) {
+      if (index === count) { return `${node.title} ${node.author}`; }
+      node = node.next_node;
+      count += 1;
+    }
+    return null;
+  }
+
+  remove(number) {
+    let node = this.head;
+    let node2 = this.head;
+    let newnode = this.head;
+    let count = 0;
+    while (node) {
+      if (number === count) {
+        newnode = node.next_node;
+      }
+      node = node.next_node;
+      count += 1;
+    }
+
+    count = 0;
+    while (node2) {
+      if (count === number - 1) {
+        node2.next_node = newnode;
+      } else if (number === 0) {
+        this.head = node2.next_node;
+        break;
+      }
+      node2 = node2.next_node;
+      count += 1;
+    }
+
+    if (this.head === undefined) {
+      this.head = null;
+      this.tail = null;
+    }
+  }
+
+  addAt(index, title, author) {
+    const newnode = new Book2(title, author);
+    let count = 0;
+    let node = this.head;
+    if (this.tail === null) {
+      this.head = newnode;
+      this.tail = newnode;
+    } else {
+      while (node) {
+        if (count === index - 1) {
+          newnode.next_node = node.next_node;
+          node.next_node = newnode;
+        } else if (index === 0) {
+          newnode.next_node = this.head;
+          this.head = newnode;
+          break;
+        }
+        count += 1;
+        node = node.next_node;
+      }
+    }
+  }
+}
+
+const lists = new BookList2();
+
 const section = document.querySelector('.bookings');
 let count = 0;
 
 function addBook(Title, Author) {
-  const booky = new Books(Title, Author);
-  bookList.push(booky);
+  const booky = new Book2(Title, Author);
+
+  lists.add(booky.title, booky.author);
 
   const container = document.createElement('div');
   const names = document.createElement('h3');
```

Comment:
- Good job adding the class 👍 however according to the requirements you should Create class methods to add and remove books. right now normal functions are handling these requests, please change that.

Reply:
This issue is well resolved! ✔️